hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStreamStatistics.java

```diff
@@ -0,0 +1,60 @@
+package org.apache.hadoop.fs.azurebfs.services;
+
+import org.apache.hadoop.classification.InterfaceStability;
+
+/**
+ * Interface for {@link AbfsOutputStream} statistics.
+ */
+@InterfaceStability.Unstable
+public interface AbfsOutputStreamStatistics {
+
+  /**
+   * Number of bytes to be uploaded.
+   *
+   * @param bytes number of bytes to upload
+   */
+  void bytesToUpload(long bytes);
+
+  /**
+   * Number of bytes uploaded Successfully.
+   *
+   * @param bytes number of bytes that were successfully uploaded
+   */
+  void bytesUploadedSuccessfully(long bytes);
+
+  /**
+   * Number of bytes failed to upload.
+   *
+   * @param bytes number of bytes that failed to upload
+   */
+  void bytesFailed(long bytes);
```

Comment:
prefer a more detailed description like uploadFailed(long). It's recording that an upload failed and the number of bytes